### ISSUE TITLE
jsdoc/valid-types reports on `MyObject[keyof MyObject]`

### ISSUE THREAD

**Issue by RobertAKARobin** [gajus/eslint-plugin-jsdoc#1357]
## Expected behavior
`MyObject[keyof MyObject]` syntax does not report.

## Actual behavior
`MyObject[keyof MyObject]` reports:

> Syntax error in type: `MyObject[keyof MyObject]` (jsdoc/valid-types)

## ESLint Config

```js
import tsEslint from 'typescript-eslint';
import jsdoc from 'eslint-plugin-jsdoc';

export default [
	{
		files: [`**/*.js`],
		languageOptions: {
			parser: tsEslint.parser,
			parserOptions: {
				ecmaFeatures: {
					latest: true,
				},
				projectService: true,
			},
		},
		plugins: {
			'@typescript-eslint': tsEslint.plugin,
			jsdoc,
		},
		rules: {
			'jsdoc/valid-types': `error`,
		},
	},
];
```

## ESLint sample

```js
/**
 * @typedef {object} MyObject
 * @typedef {MyObject[keyof MyObject]} MyObjectValue
 */
```

This also occurs in `@type`, `@param`, etc.

## Environment

- Node version: 23.6.0
- ESLint version 9.19.0
- `eslint-plugin-jsdoc` version: 50.6.3

**Comment by brettz9** [comment]
The lack of `keyof` support is an issue with our type parser; please subscribe to https://github.com/jsdoc-type-pratt-parser/jsdoc-type-pratt-parser/issues/147

**Comment by RobertAKARobin** [comment]
Thanks @brettz9 ! Is there a workaround?

**Comment by RobertAKARobin** [comment]
It looks like I can just put this in `types.d.ts` and import and use it in JSDoc:

```ts
export type PropertyOf<Type> = Type[keyof Type];
```

**Comment by brettz9** [comment]
Sure, that or use `/* eslint-disable */` directives...
